akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -340,13 +420,22 @@ final class ClusterClient(settings: ClusterClientSettings) extends Actor with Ac
       // refresh of contacts
       if (contactPoints.nonEmpty)
         contacts = contactPoints.map(context.actorSelection)
+      publishContactPoints()
     case _: ActorIdentity ⇒ // ok, from previous establish, already handled
   }
 
+  def contactPointMessages: Actor.Receive = {
+    case SubscribeContactPoints ⇒
+      sender() ! ContactPoints(contacts.map(_.anchorPath).toSet)
+      context.system.eventStream.subscribe(sender(), classOf[ContactPointChange])
```

Comment:
The reason the `Cluster` is using the `eventStream` is that it has support for subscribing to specific event classes and subclasses thereof. We don't need that here and using the `eventStream` is just confusing because you will receive events from different clients and have to filter on the receiver side. I think this should be implemented within the actor, something like (pseudo code):

```scala
  var subscribers = Vector.empty[ActorRef]

  def receive = {
    case Subscribe(ref) => 
      subscribers :+= ref
      watch(ref)
      ref ! ContactPoints(contactPaths)
    case Unsubscribe(ref) => 
      subscribers = subscribers.filterNot(_ == ref)
      unwatch(ref)
    case Terminated(ref) => 
      subscribers = subscribers.filterNot(_ == ref)
  }

  def publish(event: ContactPointChange): Unit =
    subscribers.foreach(_ ! event)
```